cluster-autoscaler/cloudprovider/azure/azure_scale_set.go

```diff
@@ -44,27 +42,6 @@ var (
 	vmssSizeMutex                     sync.Mutex
 )
 
-var scaleSetStatusCache struct {
-	lastRefresh time.Time
-	mutex       sync.Mutex
-	scaleSets   map[string]compute.VirtualMachineScaleSet
-}
-
-func init() {
-	// In go-autorest SDK https://github.com/Azure/go-autorest/blob/master/autorest/sender.go#L242,
-	// if ARM returns http.StatusTooManyRequests, the sender doesn't increase the retry attempt count,
-	// hence the Azure clients will keep retrying forever until it get a status code other than 429.
-	// So we explicitly removes http.StatusTooManyRequests from autorest.StatusCodesForRetry.
-	// Refer https://github.com/Azure/go-autorest/issues/398.
```

Comment:
@feiskyer can you confirm this code is no longer needed since https://github.com/Azure/go-autorest/issues/398 is fixed?

Reply:
yes, this is not needed after v1.18